godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,480 @@
+        let length = (*instance).inner.borrow().property_list().len();
+
+        let vec = Vec::from_raw_parts(p_prop_info as *mut GDExtensionPropertyInfo, length, length);
+
+        drop(vec);
+    }
+
+    pub(super) unsafe extern "C" fn call_func<T: ScriptInstance>(
+        p_self: GDExtensionScriptInstanceDataPtr,
+        p_method: GDExtensionConstStringNamePtr,
+        p_args: *const GDExtensionConstVariantPtr,
+        p_argument_count: GDExtensionInt,
+        r_return: GDExtensionVariantPtr,
+        r_error: *mut GDExtensionCallError,
+    ) {
+        let instance = p_self as *mut ScriptInstanceData<T>;
+        let method =
+            StringName::from_arg_ptr(p_method as GDExtensionTypePtr, PtrcallType::Standard);
+
+        let args: Vec<_> = slice::from_raw_parts(p_args, p_argument_count as usize)
+            .iter()
+            .map(|ptr| &*Variant::ptr_from_sys(force_mut_ptr(*ptr)))
+            .collect();
+
+        match (*instance)
+            .inner
+            .borrow_mut()
+            .call(method.clone(), args.as_slice())
+        {
+            Ok(ret) => {
+                ret.move_return_ptr(r_return as GDExtensionTypePtr, PtrcallType::Virtual);
+                (*r_error).error = GDEXTENSION_CALL_OK;
+            }
+
+            Err(err) => (*r_error).error = err,
+        };
+    }
+
+    pub(super) unsafe extern "C" fn get_script_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+    ) -> GDExtensionObjectPtr {
+        let instance = p_instance as *const ScriptInstanceData<T>;
+
+        let script = (*instance).inner.borrow().get_script();
+        let ptr = script.obj_sys();
+
+        mem::forget(script.into_ffi());
+
+        ptr
+    }
+
+    pub(super) unsafe extern "C" fn is_placeholder_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+    ) -> u8 {
+        let instance = p_instance as *const ScriptInstanceData<T>;
+
+        (*instance).inner.borrow().is_placeholder() as u8
+    }
+
+    pub(super) unsafe extern "C" fn has_method_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        p_method: GDExtensionConstStringNamePtr,
+    ) -> u8 {
+        let instance = p_instance as *const ScriptInstanceData<T>;
+        let method =
+            StringName::from_arg_ptr(p_method as GDExtensionTypePtr, PtrcallType::Standard);
+
+        let result = (*instance).inner.borrow().has_method(method.clone()) as u8;
+
+        result
+    }
+
+    pub(super) unsafe extern "C" fn free_method_list_func<T: ScriptInstance>(
+        p_instance: GDExtensionScriptInstanceDataPtr,
+        p_method_info: *const GDExtensionMethodInfo,
+    ) {
+        let instance = p_instance as *const ScriptInstanceData<T>;
+        let length = (*instance).inner.borrow().method_list().len();
+
+        let vec = Vec::from_raw_parts(force_mut_ptr(p_method_info), length, length);
```

Comment:
here too, this seems unsafe for the same reason as above.

Reply:
Same answer as above.